backend/core/detection.py

```diff
@@ -460,63 +460,90 @@ def handle_bgp_update(self, message: Dict) -> NoReturn:
+                                    ):
+                                        (
+                                            path_hijacker,
+                                            hij_dimensions[hij_dimension_index],
+                                        ) = func_path(
+                                            monitor_event, prefix_node, prefix_node_conf
+                                        )
+                                        if hij_dimensions[hij_dimension_index] != "-":
+                                            break
+                                elif hij_dimension_index == 2:
+                                    # data plane dimension
+                                    for func_dplane in func_dim():
+                                        hij_dimensions[
+                                            hij_dimension_index
+                                        ] = func_dplane(
+                                            monitor_event, prefix_node, prefix_node_conf
+                                        )
+                                        if hij_dimensions[hij_dimension_index] != "-":
+                                            break
+                                elif hij_dimension_index == 3:
+                                    # policy dimension
+                                    for func_pol in func_dim(
+                                        len(monitor_event["path"])
+                                    ):
+                                        (
+                                            pol_hijacker,
+                                            hij_dimensions[hij_dimension_index],
+                                        ) = func_pol(
+                                            monitor_event, prefix_node, prefix_node_conf
+                                        )
+                                        if hij_dimensions[hij_dimension_index] != "-":
+                                            break
+                                hij_dimension_index += 1
+                            # check if dimension combination in hijack combinations for this rule,
+                            # but do not commit hijack yet (record the last possible hijack issue)
+                            if hij_dimensions in HIJACK_DIM_COMBINATIONS:
+                                final_hij_dimensions = hij_dimensions[::]
+                                is_hijack = True
+                                # show pol hijacker only if the path hijacker is uncertain
+                                hijacker = path_hijacker
+                                if path_hijacker == -1 and pol_hijacker != -1:
+                                    hijacker = pol_hijacker
+                            # benign rule matching beats hijack detection
+                            else:
+                                is_hijack = False
+                                break
```

Comment:
is the break here correct? you go through the prefix node confs but you always stop on the first one if it's not a hijack?

Reply:
@slowr correct. In Artemis we follow a whitelisting logic. If a single rule renders the update innocent, it is cleared of all charges and processing should stop. Otherwise, if no innocence, I simply go through all rules and stop at the last one and use that to form the hijack (the order matters since it means that we have processed everything and we declare a hijack with the last related problem found; another solution would be to declare several hijacks but I think this is an overkill and makes things complicated without any significant gain).